src/components/ProgressButton.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { FC, useState } from 'react';
+import { View, StyleSheet, Text, TouchableOpacity } from 'react-native';
+import ProgressBar from 'react-native-progress/Bar';
+
+import { LinearGradient } from 'app/components';
+import { typography, palette } from 'app/styles';
+
+interface Props {
+  timeoutMilis: number;
+  stepIntervalMilis: number;
+  onComplete: () => any;
+  title: string;
+  inProgressTitle: string;
+}
+
+const noIntervalID = -1;
+
+export const ProgressButton: FC<Props> = ({
+  timeoutMilis,
+  stepIntervalMilis,
+  onComplete,
+  title,
+  inProgressTitle,
+}: Props) => {
+  const [inProgress, setInProgress] = useState(false);
+  const [progress, setProgress] = useState(0);
+  const [intervalID, setIntervalID] = useState(noIntervalID);
+
+  const addWallet = () => {
+    setInProgress(true);
+
+    const intervalID = setInterval(() => {
+      setIntervalID((intervalID as unknown) as number);
```

Comment:
couldn't find type to use here (or actually what to pass to `useState(` in line 27). When I log `typeof intervalID` in line 32 it s `number`. I should probably use `setInterval` from object, not straight 'from air' like I do. Checked documentation but couldn't find good answer. 
Any suggestions :)?

Reply:
![Zrzut ekranu 2021-01-19 o 16 15 10](https://user-images.githubusercontent.com/67273035/105053713-8b4fd000-5a71-11eb-91c4-eea566af9b1e.png)
no TS error in my console

Reply:
Maybe let's rename `intervalID` in the state or inside the `onPress` function? There are no conflicts or syntax errors as far as I can see but I'm not sure it's a good idea to have two variables with the same name.

Reply:
done